main.nf

```diff
@@ -15,6 +15,22 @@ nextflow.enable.dsl = 2
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
+//Either genome needs to be set or the parameters gtf (for rsem/salmon) and organism, library and keytype
+//(for pathway analysis) have to be provided separately
+//-->gtf is necessary for rsem and salmon
+if (params.input_type in ["rsem", "salmon"]) {
+    if (!params.genome && !params.gtf) { exit 1, 'Please provide either genome or gtf file!' }
+    else if (!params.gtf) { params.gtf = WorkflowMain.getGenomeAttribute(params, 'gtf') }
+}
+//-->organism, library and keytype are necessary for pathway analysis
+if (!params.skip_pathway_analysis) {
+    if (!params.genome && !params.organism) { exit 1, 'Please provide either genome or organism!' }
```

Comment:
Can you add a check here, that when `--genome` is provided,  for all fields that are needed in the `igenomes`, that it tries to load it from there, and if not available, asks the users to provide it with the respective params. I think this check is not covered here.